### PR TITLE
Make repo into gem

### LINKED ISSUE #116: User UID/GID should be set at runtime, not build time
User UID/GID should be set at runtime, not build time.  This will allow for a static build image to be re-used for multiple builds.  This is what current containers do.

_Originally posted by @treydock in https://github.com/OSC/ondemand-packaging/pull/112#discussion_r736543304_

### LINKED ISSUE #118: Static build image
Static build image - the container used to build should be static and re-usable for all OnDemand related package builds.  For RPMs there can be a single container with Mock installed. For Deb there would need to be a container per OS we support.  The deb image should use ARGS like current container to just modify which parent image gets pulled in.

### LINKED ISSUE #120: cross platform gpg signing

### LINKED ISSUE #121: build out of source tree
The building should not take place in the code source tree, but instead take tar.gz files created before build and extract them to container working directory and the built packages copied to either `dist` or some other location

### LINKED ISSUE #122: cross platform sanity checks
There should be ability to sanity check builds. Currently the RPM builds are checked to be GPG signed if GPG signing was enabled.

### LINKED ISSUE #123: cli functionality
The ability to run CLI command to do builds (can reference Ruby Rake code if we go that direction). This ensures that non-Ruby apps don't need a Rakefile and Gems as we do build many non-Ruby things.  The CLI command should take arguments that can have a help output printed, ie not using environment variables unless the help output prints possible environment variables.  It would be good to not have to read code to remember what flags / env vars are needed.

### LINKED ISSUE #127: multi task container steps.
The code executed inside the container should be able to run multiple commands because things like Mock setup and installing extra packages all require multiple commands get executed inside the container.  I would recommend use a sort of "in container" Rakefile

### PR DESCRIPTION
**Pull request by treydock**
This is missing the actual Rake task library (added #158 to track this), the CLI command is how I've been building packages to test.

Set `CONTAINER_RT=podman` to use podman

Build buildbox:

```
bundle exec rake ood_packaging:buildbox:build[el8]
```

Build a simple package:

```
 bundle exec ./bin/ood_packaging -w /tmp/work -o /tmp/output -d el8 -V 2.1 --debug $(pwd)/packages/ondemand-release
```

Fixes #127 
Fixes #123 
Fixes #122 
Fixes #120
Fixes #121 
Fixes #118 
Fixes #116 for Docker
Some fixes for #113 at least for Github Actions.

So far this was tested with Docker on Mac and podman 1.6.4 on RHEL 7.9.

The `OodPackaging::Build` class handles the actual steps inside the container to build the package.  The `OodPackaging::Package` class handles launching the containers and acts as entry point into application and `OodPackaging::BuildBox` is what handles building the images for each distribution.

### CODE REVIEW

File: .gitignore
Comment (johrstrom) [inline .gitignore:18]: Is this right?

Comment (treydock) [reply .gitignore:18]: Yep, those 3 files are ERB templates and get rendered into the container build root so they can be used during building.  They are rendered here: https://github.com/OSC/ondemand-packaging/blob/5178b63b17020c832d17249c4ea26a49543838a3/lib/ood_packaging/build_box.rb#L85-L105

**Comment by treydock** [comment]
@johrstrom Any more reviews you'd like to do on this?  I have some more changes I'd like to make but in effort to keep pull requests a little smaller, was going to wait until this was merged.

**Review by johrstrom — APPROVED** [review]
Go ahead and merge.

File: .rubocop.yml
Comment (treydock) [reply .rubocop.yml:37]: The BlockLength one is a bit overbearing but others I can enable. The issue with BlockLength is it does things like wants the OptParse block broken up as well as getting upset about almost every rspec test block.

**Pull request merged**
